### PR TITLE
Add Smithery CLI Installation Instructions

### PR DESCRIPTION
**Pull request by calclavia**
This PR adds installation instructions to automatically install Neon MCP Server for Claude Desktop using Smithery CLI. This makes it easier for users to install the MCP. Also, it adds a badge to show the number of installations from Smithery.

### CODE REVIEW

**Comment by pffigueiredo** [comment]
@calclavia, thanks for proactively adding this integration 🙏 

We will accept this PR once these are fixed on the CLI:

- the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
- the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable

**smithery CLI** ❌ 
```
{
  "mcpServers": {
    "@neondatabase-mcp-server-neon": {
      "command": "npx",
      "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
    }
  }
}
```



**neon correct config** ✅ 
```
{
  "mcpServers": {
    "neon": {
      "command": "npx",
      "args": [
        "-y",
        "@neondatabase/mcp-server-neon",
        "start",
        <NEON_API_KEY>
      ]
    }
  }
}
```

**Comment by calclavia** [comment]
> @calclavia, thanks for proactively adding this integration 🙏
> 
> We will accept this PR once these are fixed on the CLI:
> 
> * the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
> * the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable
> 
> **smithery CLI** ❌
> 
> ```
> {
>   "mcpServers": {
>     "@neondatabase-mcp-server-neon": {
>       "command": "npx",
>       "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
>     }
>   }
> }
> ```
> 
> **neon correct config** ✅
> 
> ```
> {
>   "mcpServers": {
>     "neon": {
>       "command": "npx",
>       "args": [
>         "-y",
>         "@neondatabase/mcp-server-neon",
>         "start",
>         <NEON_API_KEY>
>       ]
>     }
>   }
> }
> ```

Thanks for identifying the typo! Just fixed on our end.

Screenshot 2024-12-16 at 7 35 12 PM

claude_desktop_config.json:
```
...
    "@neondatabase-mcp-server-neon": {
      "command": "npx",
      "args": [
        "-y",
        "@neondatabase/mcp-server-neon",
        "start",
        "test"
      ]
    }
```

**Comment by pffigueiredo** [comment]
i

> > @calclavia, thanks for proactively adding this integration 🙏
> > We will accept this PR once these are fixed on the CLI:
> > 
> > * the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
> > * the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable
> > 
> > **smithery CLI** ❌
> > ```
> > {
> >   "mcpServers": {
> >     "@neondatabase-mcp-server-neon": {
> >       "command": "npx",
> >       "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
> >     }
> >   }
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > **neon correct config** ✅
> > ```
> > {
> >   "mcpServers": {
> >     "neon": {
> >       "command": "npx",
> >       "args": [
> >         "-y",
> >         "@neondatabase/mcp-server-neon",
> >         "start",
> >         <NEON_API_KEY>
> >       ]
> >     }
> >   }
> > }
> > ```
> 
> Thanks for identifying the typo! Just fixed on our end.
> 
> Screenshot 2024-12-16 at 7 35 12 PM
> 
> claude_desktop_config.json:
> 
> ```
> ...
>     "@neondatabase-mcp-server-neon": {
>       "command": "npx",
>       "args": [
>         "-y",
>         "@neondatabase/mcp-server-neon",
>         "start",
>         "test"
>       ]
>     }
> ```

I just ran the `npx @smithery/cli install @neondatabase/mcp-server-neon --client claude` command and the name of the server is still being written as `@neondatabase-mcp-server-neon` instead of just `neon` 👀 

```
{
  "mcpServers": {
    "@neondatabase-mcp-server-neon": {
      "command": "npx",
      "args": [
        "-y",
        "@neondatabase/mcp-server-neon",
        "start",
        "my-key"
      ]
    }
  }
}
```

The rest is looking great thou 💪

**Comment by calclavia** [comment]
> i
> 
> > > @calclavia, thanks for proactively adding this integration 🙏
> > > We will accept this PR once these are fixed on the CLI:
> > > 
> > > * the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
> > > * the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable
> > > 
> > > **smithery CLI** ❌
> > > ```
> > > {
> > >   "mcpServers": {
> > >     "@neondatabase-mcp-server-neon": {
> > >       "command": "npx",
> > >       "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
> > >     }
> > >   }
> > > }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > **neon correct config** ✅
> > > ```
> > > {
> > >   "mcpServers": {
> > >     "neon": {
> > >       "command": "npx",
> > >       "args": [
> > >         "-y",
> > >         "@neondatabase/mcp-server-neon",
> > >         "start",
> > >         <NEON_API_KEY>
> > >       ]
> > >     }
> > >   }
> > > }
> > > ```
> > 
> > 
> > Thanks for identifying the typo! Just fixed on our end.
> > Screenshot 2024-12-16 at 7 35 12 PM
> > claude_desktop_config.json:
> > ```
> > ...
> >     "@neondatabase-mcp-server-neon": {
> >       "command": "npx",
> >       "args": [
> >         "-y",
> >         "@neondatabase/mcp-server-neon",
> >         "start",
> >         "test"
> >       ]
> >     }
> > ```
> 
> I just ran the `npx @smithery/cli install @neondatabase/mcp-server-neon --client claude` command and the name of the server is still being written as `@neondatabase-mcp-server-neon` instead of just `neon` 👀
> 
> ```
> {
>   "mcpServers": {
>     "@neondatabase-mcp-server-neon": {
>       "command": "npx",
>       "args": [
>         "-y",
>         "@neondatabase/mcp-server-neon",
>         "start",
>         "my-key"
>       ]
>     }
>   }
> }
> ```
> 
> The rest is looking great thou 💪

Ah that's mainly because our IDs were namespaced based on your npm package ID. Claude doesn't allow slashes in the name.

Anyway, I switched it to just `neon`. It's cleaner. 😊

**Comment by calclavia** [comment]
> > > i
> > > > > @calclavia, thanks for proactively adding this integration 🙏
> > > > > We will accept this PR once these are fixed on the CLI:
> > > > > 
> > > > > * the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
> > > > > * the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable
> > > > > 
> > > > > **smithery CLI** ❌
> > > > > ```
> > > > > {
> > > > >   "mcpServers": {
> > > > >     "@neondatabase-mcp-server-neon": {
> > > > >       "command": "npx",
> > > > >       "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
> > > > >     }
> > > > >   }
> > > > > }
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > **neon correct config** ✅
> > > > > ```
> > > > > {
> > > > >   "mcpServers": {
> > > > >     "neon": {
> > > > >       "command": "npx",
> > > > >       "args": [
> > > > >         "-y",
> > > > >         "@neondatabase/mcp-server-neon",
> > > > >         "start",
> > > > >         <NEON_API_KEY>
> > > > >       ]
> > > > >     }
> > > > >   }
> > > > > }
> > > > > ```
> > > > 
> > > > 
> > > > Thanks for identifying the typo! Just fixed on our end.
> > > > Screenshot 2024-12-16 at 7 35 12 PM
> > > > claude_desktop_config.json:
> > > > ```
> > > > ...
> > > >     "@neondatabase-mcp-server-neon": {
> > > >       "command": "npx",
> > > >       "args": [
> > > >         "-y",
> > > >         "@neondatabase/mcp-server-neon",
> > > >         "start",
> > > >         "test"
> > > >       ]
> > > >     }
> > > > ```
> > > 
> > > 
> > > I just ran the `npx @smithery/cli install @neondatabase/mcp-server-neon --client claude` command and the name of the server is still being written as `@neondatabase-mcp-server-neon` instead of just `neon` 👀
> > > ```
> > > {
> > >   "mcpServers": {
> > >     "@neondatabase-mcp-server-neon": {
> > >       "command": "npx",
> > >       "args": [
> > >         "-y",
> > >         "@neondatabase/mcp-server-neon",
> > >         "start",
> > >         "my-key"
> > >       ]
> > >     }
> > >   }
> > > }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > The rest is looking great thou 💪
> > 
> > 
> > Ah that's mainly because our IDs were namespaced based on your npm package ID. Claude doesn't allow slashes in the name.
> > Anyway, I switched it to just `neon`. It's cleaner. 😊
> 
> Ah, I see!
> 
> Ideally it would just be `neon`, but it that doesn't work for you at all, I think we are OK with having "@neondatabase-mcp-server-neon" as the `key` of the MCP Server instead, this last update doesn't work because `neon` isn't a pacakge that `npx` can use 😓
> 
> So let's do one of these:
> 
> 1- Rollback to the previous version and just use "@neondatabase-mcp-server-neon" as the `key` in the `mcpServers` JSON 2- Use "neon" as the `key`, but keep "@neondatabase/mcp-server-neon" as the name of the package
> 
> Option 2 is preferred, but we can go with option 1, if that's too hard to get right on your side 👍
> 
> Sorry for all the back and forth, didn't understand that the name of the server was coupled with the name of the package!

Oh, the ID of the installation on Smithery is not coupled with the NPM package. I just named it based on how you defined it in the package, so no changes are needed and it should just work now.

**Comment by pffigueiredo** [comment]
> @neondatabase/mcp-server-neon

So, I think we got a bit confused here, this is the output of `npx -y @smithery/cli install neon --client claude`:

```json
{
  "mcpServers": {
    "neon": {
      "command": "npx",
      "args": ["-y", "neon", "start", "API_KEY"]
    }
  }
}

```

And this is what it should be:
```json
 "neon": {
      "command": "npx",
      "args": [
        "-y",
        "@neondatabase/mcp-server-neon",
        "start",
        "API_KEY"
      ]
    }
```


The only difference right now is just the name of the package that `npx` should pick up

**Comment by calclavia** [comment]
> > @neondatabase/mcp-server-neon
> 
> So, I think we got a bit confused here, this is the output of `npx -y @smithery/cli install neon --client claude`:
> 
> ```json
> {
>   "mcpServers": {
>     "neon": {
>       "command": "npx",
>       "args": ["-y", "neon", "start", "API_KEY"]
>     }
>   }
> }
> ```
> 
> And this is what it should be:
> 
> ```json
>  "neon": {
>       "command": "npx",
>       "args": [
>         "-y",
>         "@neondatabase/mcp-server-neon",
>         "start",
>         "API_KEY"
>       ]
>     }
> ```
> 
> The only difference right now is just the name of the package that `npx` should pick up

Fixed this. Now it correctly sets `@neondatabase/mcp-server-neon` while using `neon` as the name

Screenshot 2024-12-17 at 10 55 23 AM

**Comment by pffigueiredo** [comment]
> > > @neondatabase/mcp-server-neon
> > 
> > 
> > So, I think we got a bit confused here, this is the output of `npx -y @smithery/cli install neon --client claude`:
> > ```json
> > {
> >   "mcpServers": {
> >     "neon": {
> >       "command": "npx",
> >       "args": ["-y", "neon", "start", "API_KEY"]
> >     }
> >   }
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And this is what it should be:
> > ```json
> >  "neon": {
> >       "command": "npx",
> >       "args": [
> >         "-y",
> >         "@neondatabase/mcp-server-neon",
> >         "start",
> >         "API_KEY"
> >       ]
> >     }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The only difference right now is just the name of the package that `npx` should pick up
> 
> Fixed this. Now it correctly sets `@neondatabase/mcp-server-neon` while using `neon` as the name
> 
> Screenshot 2024-12-17 at 10 55 23 AM

Awesome, I just end-to-end tested and it's working great  💪 

Thanks a lot for getting this though 🚀

File: README.md [hunk 1]
```diff
@@ -10,6 +12,14 @@ This lets you use Claude Desktop, or any MCP Client, to use natural language to
 
 # Claude Setup
 
+### Installing via Smithery
+
+To install Neon MCP Server for Claude Desktop automatically via [Smithery](https://smithery.ai/protocol/neon):
+
+```bash
+npx @smithery/cli install @neondatabase/mcp-server-neon --client claude
```
Comment (pffigueiredo) [inline README.md:20]:
```suggestion
npx -y @smithery/cli install neon --client claude
```

**Review by pffigueiredo — APPROVED** [review]
Looks great, will proceed with merging this into upstream now 🚀

**Pull request merged**